docs/agent/config.md

```diff
@@ -87,7 +87,191 @@ differently from Agent version 5.
+
+Example:
+
+```
+instances:
+  - server: %%host%%
+    user: ENC[db_prod_user_%%host%%]
+    password: ENC[db_prod_password_%%host%%]
+```
+
+### Fetching secrets
+
+To fetch the secrets from you configurations you have to provide an executable
+capable of fetching passwords from your secrets management tool.
+
+An external executable solved multiple issues:
+
+- Guarantees that the agent will never have more access/information than what
+  you transfer to it.
+- Maximum freedom and flexibility: allowing users to use any secrets management
+  tool (including closed sources ones) without having to rebuild the agent.
+- Solving the **initial trust** is a very complex problem as it usually changes
+  from users to users. Each setup requires more or less control and might rely on
+  private tools. This shifts the problem to where it is easier to solve, without
+  having to rebuild the agent.
+
+This executable will be executed by the agent once per check's instance. The
+agent will cache secrets internally to reduce the number of calls (useful in
+a containerized environment for example).
+
+The executable **MUST** (the agent will refuse to use it otherwise):
+
+- Belong to the same user running the agent (usually `dd-agent`).
+- Have **no** rights for `group` or `other`.
+- Have at least `exec` right for the owner.
+- The executable will not share any environment variables with the agent.
+
+#### Configuration
+
+In `datadog.yaml` you must set the following variables:
+
+```yaml
+secret_backend_command: /path/to/you/executable
+```
+
+More settings are available: see `datadog.yaml`.
+
+#### The executable API
+
+The executable has to respect a very simple API: it reads a JSON on the
+Standard input and output a JSON containing the decrypted secrets on the
+Standard output.
+
+**Input:**
+
+The executable will receive a JSON payload on the `Standard input` containing
+the list of secret to fetch:
+
+```json
+{
+  "version": "1.0",
+  "secrets": ["secret1", "secret2"]
+}
+```
+
+- `version`: is a string containing the format version (currently "1.0").
+- `secrets`: is a list of strings, each string is a **handle** from a
+  configuration corresponding to a secret to fetch.
+
+**Output:**
+
+The executable is expected to output on the `Standard output` a JSON containing
+the fetched secrets:
+
+```json
+{
+  "secret1": {
+    "value": "secret_value",
+    "error": null
+  },
+  "secret2": {
+    "value": null,
+    "error": "could not fetch the secret"
+  }
+}
+```
+
+The expected payload is a JSON object, each key is one of the **handle** requested in the input payload.
+The value for each **handle** is a JSON object with 2 fields:
+
+- `value`: a string: the actual secret value to be used in the check
+  configurations (can be `null` in case of error).
+- `error`: a string: the error message if needed. If `error` is different that
+  `null` the configuration will be considered erroneous and dropped.
```

Comment:
I'd say `the integration configuration that uses this handle will be [...]` instead here :)